File: .sdk/tm/php/utility/PrepareHeaders.php
<?php
declare(strict_types=1);

// WeatherDataApi2 SDK utility: prepare_headers

class WeatherDataApi2PrepareHeaders
{
    public static function call(WeatherDataApi2Context $ctx): array
    {
        $options = $ctx->client->options_map();
        $headers = \Voxgig\Struct\Struct::getprop($options, 'headers');
        if (!$headers) {
            return [];
        }
        $out = \Voxgig\Struct\Struct::clone($headers);
        return is_array($out) ? $out : [];
    }
}

File: .sdk/tm/rb/utility/prepare_headers.rb
# WeatherDataApi2 SDK utility: prepare_headers
require_relative 'struct/voxgig_struct'
module WeatherDataApi2Utilities
  PrepareHeaders = ->(ctx) {
    options = ctx.client.options_map
    headers = VoxgigStruct.getprop(options, "headers")
    return {} unless headers
    out = VoxgigStruct.clone(headers)
    out.is_a?(Hash) ? out : {}
  }
end
